### PR TITLE
[cppmicroservices] Add new port (fix #5801)

### LINKED ISSUE #5801: Make CppMicroservices installable via vcpkg
Hi everyone,

I would like to ask if there is the possibility to be able to install CppMicroservices with the command line utility vcpkg which makes package installation really simple.

an example use case would be on the command line:

vcpkg install cppmicroservices

Project repo : https://github.com/CppMicroServices/CppMicroServices

Feature request : https://github.com/CppMicroServices/CppMicroServices/issues/329

### PR DESCRIPTION
**Pull request by saschazelzer**
I tested this port on Windows using

x86-windows
x86-windows-static
x64-windows
x64-windows-static

and Linux (Fedora 29)

x64-linux
x64-linux-static

I do not have access to a Windows ARM platform though.

### CODE REVIEW

**Comment by NancyLi1013** [comment]
Hi @saschazelzer, thanks for the new port. I see some failures from the current CI system:

|x64-osx         |master| test |notes
|----------------|------|------|-----
|cppmicroservices|      |Fail  |New Port

|arm64-windows   |master| test |notes
|----------------|------|------|-----
|cppmicroservices|      |Fail  |New Port

|x64-uwp         |master| test |notes
|----------------|------|------|-----
|cppmicroservices|      |Fail  |New Port

|arm-uwp         |master| test |notes
|----------------|------|------|-----
|cppmicroservices|      |Fail  |New Port

Please refer to the following attachment to see what happended on these triplets.
[failureLogs.zip](https://github.com/microsoft/vcpkg/files/3176294/failureLogs.zip)

x64-windows-static results are inconclusive because it is missing logs from test
**As for x64-windows-static, we get some exception info like this:**

-- Performing post-build validation
Running command:
   "C:\VS2017\Stable\VC\Tools\MSVC\14.16.27023\bin\HostX86\x86\dumpbin.exe" /directives "C:\vsts\_work\2\s\packages\cppmicroservices_x64-windows-static\debug\lib\CppMicroServices4d.lib"
 failed with message:
Microsoft (R) COFF/PE Dumper Version 14.16.27026.1
Dump of file C:\vsts\_work\2\s\packages\cppmicroservices_x64-windows-static\debug\lib\CppMicroServices4d.lib

C:\vsts\_work\2\s\packages\cppmicroservices_x64-windows-static\debug\lib\CppMicroServices4d.lib : fatal error LNK1107: invalid or corrupt file: cannot read at 0x9EA2F92

File: ports/cppmicroservices/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+vcpkg_install_cmake()
+
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/share)
+
+# Handle copyright
+file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/cppmicroservices RENAME copyright)
+
+# Post-build test for cmake libraries
+vcpkg_test_cmake(PACKAGE_NAME CppMicroServices)
```
Comment (ras0219-msft) [inline ports/cppmicroservices/portfile.cmake]: `vcpkg_fixup_cmake_targets()`?

**Comment by dan-shaw** [comment]
Thanks for the PR!

**Pull request merged**